Packs/iDefense/Integrations/iDefense_v2/iDefense_v2.py

```diff
@@ -0,0 +1,274 @@
+from typing import Union
+
+from CommonServerPython import *
+
+'''IMPORTS'''
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+'''CONSTANTS'''
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+
+
+class Client(BaseClient):
+
+    def __init__(self, input_url: str, api_key: str, verify_certificate: bool, proxy: bool):
+        base_url = urljoin(input_url, '/rest/threatindicator/v0')
+        headers = {
+            "Content-Type": "application/json",
+            'auth-token': api_key
+        }
+        super(Client, self).__init__(base_url=base_url,
+                                     verify=verify_certificate,
+                                     headers=headers,
+                                     proxy=proxy)
+
+    def perform_request(self, name: str, value: str):
+        data = {}
+        if value:
+            data = {'key.values': value}
+        return self._http_request(method='GET', url_suffix=name, params=data)
+
+
+def _validate_args(indicator_type: str, value: str):
+    """
+    Args:
+        indicator_type: IP or URL
+        value: indicator value
+
+    Returns: True if the value is valid for use
+
+    """
+    is_valid = False
+    if indicator_type == 'IP':
+        if re.match(ipv4Regex, value):
+            is_valid = True
+    if indicator_type == 'URL':
+        if re.match(urlRegex, value):
+            is_valid = True
+    return is_valid
+
+
+def _calculate_dbot_score(severity: int):
+    """
+    Calculates Dbot score according to table:
+    Dbot Score   | severity
+     0           | 0
+     1           | 1,2
+     2           | 3,4
+     3           | 5,6,7
+    Args:
+        severity: value from 1 to 5, determined by iDefense threat indicator
+
+    Returns:
+        Calculated score
+    """
+
+    if severity > 4:
+        dbot_score = Common.DBotScore.BAD
+    elif severity > 2:
+        dbot_score = Common.DBotScore.SUSPICIOUS
+    elif severity > 0:
+        dbot_score = Common.DBotScore.GOOD
+    else:
+        dbot_score = Common.DBotScore.NONE
+    return dbot_score
+
+
+def _extract_analysis_info(res: dict, indicator_value: str, dbot_score_type: str):
+    """
+    Extract context data from http-response and create corresponding DBotScore.
+    If response is empty, return empty context and a none for DBotScore object
+    Args:
+        res: response from http request
+        indicator_value: value of indicator given as calling the command
+        dbot_score_type: DBotScoreType
+
+    Returns:
+        analysis_info: dictionary contains the indicator details returned
+        dbot: DBotScore regarding the specific indicator
+    """
+    dbot = None
+    analysis_info = {}
+    if res.get('total_size'):
+        results_array = res.get('results', [])
+        if len(results_array):
+            result_content = results_array[0]
```

Comment:
is it always an arr of length 1?

Reply:
sometimes it can be more, I saw array of 2 but the second was with the same uuid. If you remember we talked about it that the result is indicator, so one result is the relevant

Reply:
yes I remember now, so please add an inline in the code for the next dev to come.